### PR TITLE
[YHS-165] upadte build and add command in chart

### PR DESCRIPTION
**Pull request by 0yukali0**
# What is PR?
This PR fixs the failure by adopts CGO option when compiling and adds the command to the chart/template/deployment.yml. 

# Related issues
#165 

# How to validate?
```
# prepare a kind cluster
make kind-all
# load custom image to the kind
make kind-load-image
sed -i 's/tag: .*/tag: "<YOUR IMAGE TAG>"/g' charts/yunikorn-history-server/values.yaml
helm install yhs chart -n yunikorn
```
# Screenshots
image
image

### CODE REVIEW

**Comment by 0yukali0** [comment]
#165

File: Makefile [hunk 1]
```diff
@@ -274,7 +274,7 @@ web-build: ng ## build the web components.
 .PHONY: build
 build: bin/app ## build the yunikorn-history-server binary for current OS and architecture.
 	echo "Building yunikorn-history-server binary for $(OS)/$(ARCH)"
-	GOOS=$(OS) GOARCH=$(ARCH) $(GO) build -o $(LOCALBIN_APP)/yunikorn-history-server 										\
+	CGO_ENABLED=0 GOOS=$(OS) GOARCH=$(ARCH) $(GO) build -o $(LOCALBIN_APP)/yunikorn-history-server 										\
```
Comment (sudiptob2) [inline Makefile:277]: Hi, What is the particular reason for disabling CGO? I don't think it is related to the failure. Could you provide a little bit more details please?

File: charts/yunikorn-history-server/templates/deployment.yaml [hunk 2]
```diff
@@ -39,6 +39,7 @@ spec:
         - name: "yunikorn-history-server"
           image: "{{ include "yunikorn-history-server.image" . }}"
           imagePullPolicy: "{{ $imagePullPolicy }}"
+          command: ["/app/yunikorn-history-server"]
```
Comment (sudiptob2) [inline charts/yunikorn-history-server/templates/deployment.yaml:42]: We already have this command to the `Entrypoint` of our docker image. Therefore, we do not need this.
```suggestion
```

**Review by sudiptob2 — CHANGES_REQUESTED** [review]
Hi, Thanks for the contribution 🚀 
I think the only problem with the helm deployment is wrong cluster name. The variable `KIND_CLUSTER` does not exist. Therefore we have to use `CLUSTER_NAME` variable.
Nice catch on that 🙏

File: Makefile (hunk 1 above)
Comment (0yukali0) [reply Makefile:277]: Hi @sudiptob2, thanks for review.
Binary file in the alpine is not executable without disabling CGO. After setting this option, it is executable.
Current version.
image
image
Modified version.
image
[Ref](https://stackoverflow.com/questions/36279253/go-compiled-binary-wont-run-in-an-alpine-docker-container-on-ubuntu-host)

File: charts/yunikorn-history-server/templates/deployment.yaml (hunk 2 above)
Comment (0yukali0) [reply charts/yunikorn-history-server/templates/deployment.yaml:42]: Agree.
I will remove this in next commit.

File: Makefile (hunk 1 above)
Comment (0yukali0) [reply Makefile:277]: Hi @sudiptob2. 
My machine setting is following.
Description:    Ubuntu 22.04.3 LTS
Release:        22.04
Arch:  x86_64

Comment (richscott) [reply Makefile:277]: @0yukali0 @sudiptob2 I tested on my Ubuntu 22.04.4 LTS system (x86_64 arch) and was seeing the error - it does not build the `bin/app/yunikorn-history-server` - the solution of using `CGO_ENABLED=0` does work. I found out from reading

  https://stackoverflow.com/questions/61515186/when-using-cgo-enabled-is-must-and-what-happens

that CGO is disabled by default when cross-compiling (which is why we didn't see it when doing our builds on macOS/arm64) but enabled by default on native compilations. I think this PR is good.

**Review by richscott — APPROVED** [review]

**Review by sudiptob2 — APPROVED** [review]
`CGO` isn't strictly required for us, so disabling it by default seems the most user-friendly option.

**Pull request merged**